src/internal/scheduler.js

```diff
@@ -14,23 +14,25 @@ let semaphore = 0
   and flushed after this task has finished (assuming the scheduler endup in a released
   state).
 **/
-function exec(task) {
-  try {
-    suspend()
-    task()
-  } finally {
-    flush()
+function exec() {
+  let task;
+  while (!semaphore && (task = queue.shift()) !== undefined) {
```

Comment:
Actually flush and exec can be combined in a single method (processQueue for example).